docs/current/guides/324301-argo-workflows.md

```diff
@@ -0,0 +1,110 @@
+
+```yaml file=./snippets/argo-workflows/daemonset.yaml
+```
+
+- `registry.dagger.io/engine:{VERSION}` is specified twice. Ensure it has the appropriate version for your use case.
+- `buildkit-v0.12.1.linux-amd64.tar.gz` is downloaded to use for the readinessProbe. If you're running on arm64, be sure to change this to `buildkit-v0.12.1.linux-arm64.tar.gz`.
+- `/var/run/buildkit` is configured as a volume mount. This is how the workflow pods communicate with the engine.
+
+When you're satisfied with the configuration, apply the daemonset
+
+`kubectl apply -f ./daemonset.yaml`
+
+## Run a sample workflow
+
+The sample workflow will clone and run the CI for the [greetings-api](https://github.com/kpenfound/greetings-api) demo project. This project uses the Dagger Go SDK for CI.
+
+Create a file called `workflow.yaml` with the following content:
+
+```yaml file=./snippets/argo-workflows/workflow.yaml
+```
+
+- The workflow uses hardwired artifacts to clone the git repository and to install the Dagger CLI
+- `/var/run/buildkit` is mounted and specified with the `_EXPERIMENTAL_DAGGER_RUNNER_HOST` environment variable
+- The Dagger CLI `dagger_v0.8.4_linux_amd64.tar.gz` is downloaded and installed. Confirm the version and architecture are accurate for your cluster and project
+- The image `golang:1.21.0-bookworm` is used as the runtime for the container because the example project requires Go
+- The environment variable `_EXPERIMENTAL_DAGGER_CLOUD_TOKEN` is set from the kubernetes secret `dagger-cloud.token`. If you have a dagger cloud token, set this as a secret with `kubectl create secret generic dagger-cloud --from-literal=token={YOUR_TOKEN}`. If not, remove this variable.
+
+The workflow uses a persistentVolumeClaim for the runtime dependencies of the pipeline. Even though the dependencies within the pipeline are cached through Dagger, we still have dependencies to run the pipeline. Create the persistentVolumeClaim configuration in a file called gomodcache.yaml:
+
+```yaml file=./snippets/argo-workflows/gomodcache.yaml
+```
+
+and apply the persistentVolumeClaim:
+
+ `kubectl apply -f ./gomodcache.yaml`
+
+When you're satisfied with the workflow configuration, run it with argo
+
+`argo submit -n argo --watch ./workflow.yaml`
+
+With `--watch` you'll see the status of the workflow request in Argo. If you want to see the logs from your workflow, note the pod name and in another terminal run `kubectl logs -f {POD_NAME}`
+
+Once the workflow has successfully completed, run it again with `argo submit -n argo --watch ./workflow.yaml` and notice the caching this time!
+
+## Closing thoughts
+
+This example demonstrated a very basic way to run Dagger with Argo Workflows. In real world environments, Argo Workflows is often coupled with ArgoCD and Argo Events to create a fully featured CI/CD platform. Read more about that [here](https://medium.com/atlantbh/implementing-ci-cd-pipeline-using-argo-workflows-and-argo-events-6417dd157566).
```

Comment:
```suggestion
This example demonstrated how to integrate Dagger with Argo Workflows. Use the [API Key Concepts](../api/975146-concepts.mdx) page and the [Go](https://pkg.go.dev/dagger.io/dagger), [Node.js](../sdk/nodejs/reference/modules.md) and [Python](https://dagger-io.readthedocs.org/) SDK References to learn more about Dagger. 

For more information on Argo Workflows, refer to the [official documentation](https://argoproj.github.io/argo-workflows/#documentation) or to third-party tutorials such as this [guide on implementing CI/CD pipeline using Argo Workflows and Argo Events](https://medium.com/atlantbh/implementing-ci-cd-pipeline-using-argo-workflows-and-argo-events-6417dd157566).
```

Reply:
I think this says something slightly different than what I was trying to say. The point is that the argo-workflows deployment in the guide isn't a very realistic setup, and in real environments argo-workflows will be paired with argocd and argo-events

Reply:
Thanks for clarifying. I've modified my suggestion, see if better?